pkg/nats/factory.go

```diff
@@ -0,0 +1,16 @@
+package nats
+
+import (
+	"strings"
+)
+
+func GetMarshaler(format string) MarshalerUnmarshaler {
+	switch strings.ToLower(format) {
+	case "nats-core":
+		return &NATSMarshaler{}
+	case "json":
+		return &JSONMarshaler{}
+	default:
+		return &GobMarshaler{}
```

Comment:
What do you think about using NATSMarshaler by default? Gob may be a bit fragile for production usage (due to some binary incompatibilities and harder debugging: binary data is pretty hard to read by human 😅)

Reply:
this is done see https://github.com/AlexCuse/watermill-nats/commit/282a1f46f8530b367b4b9685a0fd57269abc4d64

Reply:
👌